### ISSUE TITLE
💅 `useMathMinMax` false positive with Date values

### ISSUE THREAD

**Issue by quiqueg** [biomejs/biome#10272]
### Environment information

<details>

```bash
CLI:
  Version:                      2.4.14
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:                    unset
  BIOME_LOG_PREFIX_NAME:             unset
  BIOME_LOG_LEVEL:                   unset
  BIOME_LOG_KIND:                    unset
  BIOME_CONFIG_PATH:                 unset
  BIOME_THREADS:                     unset
  BIOME_WATCHER_KIND:                unset
  BIOME_WATCHER_POLLING_INTERVAL:    unset
  NO_COLOR:                     unset
  TERM:                         xterm
  JS_RUNTIME_VERSION:           v24.14.1
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         pnpm/10.33.2

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         /workspace/biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true
  HTML full support enabled:    unset

Linter:
  JavaScript enabled:           unset
  JSON enabled:                 unset
  CSS enabled:                  unset
  GraphQL enabled:              unset
  Recommended:                  true
  Enabled rules:
    a11y/noAccessKey
    a11y/noAriaHiddenOnFocusable
    a11y/noAriaUnsupportedElements
    a11y/noAutofocus
    a11y/noDistractingElements
    a11y/noHeaderScope
    a11y/noInteractiveElementToNoninteractiveRole
    a11y/noLabelWithoutControl
    a11y/noNoninteractiveElementToInteractiveRole
    a11y/noNoninteractiveTabindex
    a11y/noPositiveTabindex
    a11y/noRedundantAlt
    a11y/noRedundantRoles
    a11y/noStaticElementInteractions
    a11y/noSvgWithoutTitle
    a11y/useAltText
    a11y/useAnchorContent
    a11y/useAriaActivedescendantWithTabindex
    a11y/useAriaPropsForRole
    a11y/useAriaPropsSupportedByRole
    a11y/useButtonType
    a11y/useFocusableInteractive
    a11y/useGenericFontNames
    a11y/useHeadingContent
    a11y/useHtmlLang
    a11y/useIframeTitle
    a11y/useKeyWithClickEvents
    a11y/useKeyWithMouseEvents
    a11y/useMediaCaption
    a11y/useSemanticElements
    a11y/useValidAnchor
    a11y/useValidAriaProps
    a11y/useValidAriaRole
    a11y/useValidAriaValues
    a11y/useValidAutocomplete
    a11y/useValidLang
    complexity/noAdjacentSpacesInRegex
    complexity/noArguments
    complexity/noBannedTypes
    complexity/noCommaOperator
    complexity/noEmptyTypeParameters
    complexity/noExcessiveNestedTestSuites
    complexity/noExtraBooleanCast
    complexity/noFlatMapIdentity
    complexity/noImportantStyles
    complexity/noStaticOnlyClass
    complexity/noThisInStatic
    complexity/noUselessCatch
    complexity/noUselessConstructor
    complexity/noUselessContinue
    complexity/noUselessEmptyExport
    complexity/noUselessEscapeInRegex
    complexity/noUselessFragments
    complexity/noUselessLabel
    complexity/noUselessLoneBlockStatements
    complexity/noUselessRename
    complexity/noUselessStringRaw
    complexity/noUselessSwitchCase
    complexity/noUselessTernary
    complexity/noUselessThisAlias
    complexity/noUselessTypeConstraint
    complexity/noUselessUndefinedInitialization
    complexity/useArrowFunction
    complexity/useDateNow
    complexity/useFlatMap
    complexity/useIndexOf
    complexity/useLiteralKeys
    complexity/useNumericLiterals
    complexity/useOptionalChain
    complexity/useRegexLiterals
    complexity/useSimpleNumberKeys
    correctness/noConstAssign
    correctness/noConstantCondition
    correctness/noConstantMathMinMaxClamp
    correctness/noConstructorReturn
    correctness/noEmptyCharacterClassInRegex
    correctness/noEmptyPattern
    correctness/noGlobalObjectCalls
    correctness/noInnerDeclarations
    correctness/noInvalidBuiltinInstantiation
    correctness/noInvalidConstructorSuper
    correctness/noInvalidDirectionInLinearGradient
    correctness/noInvalidGridAreas
    correctness/noInvalidPositionAtImportRule
    correctness/noInvalidUseBeforeDeclaration
    correctness/noMissingVarFunction
    correctness/noNonoctalDecimalEscape
    correctness/noPrecisionLoss
    correctness/noPrivateImports
    correctness/noSelfAssign
    correctness/noSetterReturn
    correctness/noStringCaseMismatch
    correctness/noSwitchDeclarations
    correctness/noUndeclaredDependencies
    correctness/noUndeclaredVariables
    correctness/noUnknownFunction
    correctness/noUnknownMediaFeatureName
    correctness/noUnknownProperty
    correctness/noUnknownPseudoClass
    correctness/noUnknownPseudoElement
    correctness/noUnknownTypeSelector
    correctness/noUnknownUnit
    correctness/noUnmatchableAnbSelector
    correctness/noUnreachable
    correctness/noUnreachableSuper
    correctness/noUnsafeFinally
    correctness/noUnsafeOptionalChaining
    correctness/noUnusedFunctionParameters
    correctness/noUnusedImports
    correctness/noUnusedLabels
    correctness/noUnusedPrivateClassMembers
    correctness/noUnusedVariables
    correctness/noVoidElementsWithChildren
    correctness/noVoidTypeReturn
    correctness/useGraphqlNamedOperations
    correctness/useIsNan
    correctness/useParseIntRadix
    correctness/useSingleJsDocAsterisk
    correctness/useValidForDirection
    correctness/useValidTypeof
    correctness/useYield                                                                         nursery/noConditionalExpect                                                                  nursery/noDeprecatedMediaType                                                                nursery/noDivRegex
    nursery/noDuplicateAttributes                                                                nursery/noDuplicateEnumValues                                                                nursery/noDuplicateSelectors
    nursery/noEmptyObjectKeys                                                                    nursery/noExcessiveNestedCallbacks                                                           nursery/noFloatingClasses                                                                    nursery/noForIn                                                                              nursery/noIdenticalTestTitle                                                                 nursery/noImpliedEval                                                                        nursery/noJsxLeakedDollar                                                                    nursery/noLoopFunc                                                                           nursery/noMisleadingReturnType                                                               nursery/noMisusedPromises                                                                    nursery/noMultiAssign                                                                        nursery/noMultiStr                                                                           nursery/noNestedPromises                                                                     nursery/noParametersOnlyUsedInRecursion                                                      nursery/noProto                                                                              nursery/noReactStringRefs                                                                    nursery/noRedundantDefaultExport                                                             nursery/noReturnAssign                                                                       nursery/noScriptUrl                                                                          nursery/noSyncScripts                                                                        nursery/noTopLevelLiterals                                                                   nursery/noUnnecessaryTemplateExpression                                                      nursery/noUnsafePlusOperands                                                                 nursery/noUselessReturn                                                                      nursery/useArraySome                                                                         nursery/useArraySortCompare                                                                  nursery/useConsistentEnumValueType                                                           nursery/useConsistentMethodSignatures                                                        nursery/useConsistentTestIt                                                                  nursery/useDestructuring                                                                     nursery/useDisposables                                                                       nursery/useDomNodeTextContent                                                                nursery/useDomQuerySelector                                                                  nursery/useErrorCause                                                                        nursery/useExpect                                                                            nursery/useFind                                                                              nursery/useGlobalThis                                                                        nursery/useImportsFirst                                                                      nursery/useMathMinMax                                                                        nursery/useNamedCaptureGroup                                                                 nursery/useRegexpTest                                                                        nursery/useTestHooksOnTop                                                                    nursery/useUnicodeRegex                                                                      performance/noAccumulatingSpread                                                             performance/noAwaitInLoops                                                                   performance/noDynamicNamespaceImportAccess                                                   security/noBlankTarget                                                                       security/noGlobalEval                                                                        style/noCommonJs                                                                             style/noDescendingSpecificity                                                                style/noMagicNumbers                                                                         style/noNamespace
    style/noNonNullAssertion
    style/noProcessEnv
    style/noRestrictedImports
    style/useArrayLiterals
    style/useAsConstAssertion
    style/useConsistentTypeDefinitions
    style/useConst
    style/useDeprecatedReason
    style/useExponentiationOperator
    style/useExportType
    style/useGroupedAccessorPairs
    style/useImportType
    style/useLiteralEnumMembers
    style/useNodejsImportProtocol
    style/useObjectSpread
    style/useReadonlyClassProperties
    style/useShorthandFunctionType
    style/useSymbolDescription
    style/useTemplate
    style/useUnifiedTypeSignatures
    suspicious/noApproximativeNumericConstant
    suspicious/noAssignInExpressions
    suspicious/noAsyncPromiseExecutor
    suspicious/noBiomeFirstException
    suspicious/noBitwiseOperators
    suspicious/noCatchAssign
    suspicious/noClassAssign
    suspicious/noCommentText
    suspicious/noCompareNegZero
    suspicious/noConfusingLabels
    suspicious/noConfusingVoidType
    suspicious/noConstEnum
    suspicious/noConstantBinaryExpressions
    suspicious/noControlCharactersInRegex
    suspicious/noDebugger
    suspicious/noDocumentCookie
    suspicious/noDoubleEquals
    suspicious/noDuplicateAtImportRules
    suspicious/noDuplicateCase
    suspicious/noDuplicateClassMembers
    suspicious/noDuplicateCustomProperties
    suspicious/noDuplicateDependencies
    suspicious/noDuplicateElseIf
    suspicious/noDuplicateFields
    suspicious/noDuplicateFontNames
    suspicious/noDuplicateJsxProps
    suspicious/noDuplicateObjectKeys
    suspicious/noDuplicateParameters
    suspicious/noDuplicateProperties
    suspicious/noDuplicateSelectorsKeyframeBlock
    suspicious/noDuplicateTestHooks
    suspicious/noEmptyBlock
    suspicious/noEmptyBlockStatements
    suspicious/noEmptyInterface
    suspicious/noEmptySource
    suspicious/noExplicitAny
    suspicious/noExportsInTest
    suspicious/noExtraNonNullAssertion
    suspicious/noFallthroughSwitchClause
    suspicious/noFocusedTests
    suspicious/noFunctionAssign
    suspicious/noGlobalAssign
    suspicious/noGlobalIsFinite
    suspicious/noGlobalIsNan
    suspicious/noImplicitAnyLet
    suspicious/noImportAssign
    suspicious/noImportantInKeyframe
    suspicious/noIrregularWhitespace
    suspicious/noLabelVar
    suspicious/noMisleadingCharacterClass
    suspicious/noMisleadingInstantiator
    suspicious/noMisrefactoredShorthandAssign
    suspicious/noNonNullAssertedOptionalChain
    suspicious/noOctalEscape
    suspicious/noPrototypeBuiltins
    suspicious/noQuickfixBiome
    suspicious/noRedeclare
    suspicious/noRedundantUseStrict
    suspicious/noSelfCompare
    suspicious/noShadowRestrictedNames
    suspicious/noShorthandPropertyOverrides
    suspicious/noSparseArray
    suspicious/noSuspiciousSemicolonInJsx
    suspicious/noTemplateCurlyInString
    suspicious/noThenProperty
    suspicious/noTsIgnore
    suspicious/noUnassignedVariables
    suspicious/noUnknownAtRules
    suspicious/noUnsafeDeclarationMerging
    suspicious/noUnsafeNegation
    suspicious/noUnusedExpressions
    suspicious/noUselessEscapeInString
    suspicious/noUselessRegexBackrefs
    suspicious/noVar
    suspicious/noWith
    suspicious/useAdjacentOverloadSignatures
    suspicious/useBiomeIgnoreFolder
    suspicious/useDefaultSwitchClauseLast
    suspicious/useGetterReturn
    suspicious/useGoogleFontDisplay
    suspicious/useIsArray
    suspicious/useIterableCallbackReturn
    suspicious/useNamespaceKeyword
    suspicious/useStaticResponseMethods

Workspace:
  Open Documents:               0
```

</details>

### Rule name

useMathMinMax

### Playground link

https://biomejs.dev/playground/?lintRules=useMathMinMax&tab=analyzer-fixes&pane=Diagnostics&code=YwBvAG4AcwB0ACAAZABhAHQAZQBzADoAIABEAGEAdABlAFsAXQAgAD0AIABbAF0AOwAKAAoAZQB4AHAAbwByAHQAIABjAG8AbgBzAHQAIABsAGEAdABlAHMAdABEAGEAdABlACAAPQAgAGQAYQB0AGUAcwAuAHIAZQBkAHUAYwBlADwARABhAHQAZQAgAHwAIAB1AG4AZABlAGYAaQBuAGUAZAA%2BACgACgAgACAAKABsAGEAdABlAHMAdAAsACAAYwB1AHIAcgBlAG4AdAApACAAPQA%2BACAAewAKACAAIAAgACAAaQBmACAAKAAhAGwAYQB0AGUAcwB0ACkAIAB7AAoAIAAgACAAIAAgACAAcgBlAHQAdQByAG4AIABjAHUAcgByAGUAbgB0ADsACgAgACAAIAAgAH0ACgAgACAAIAAgAGkAZgAgACgAIQBjAHUAcgByAGUAbgB0ACkAIAB7AAoAIAAgACAAIAAgACAAcgBlAHQAdQByAG4AIABsAGEAdABlAHMAdAA7AAoAIAAgACAAIAB9AAoAIAAgACAAIAByAGUAdAB1AHIAbgAgAGwAYQB0AGUAcwB0ACAAPgAgAGMAdQByAHIAZQBuAHQAIAA%2FACAAbABhAHQAZQBzAHQAIAA6ACAAYwB1AHIAcgBlAG4AdAA7AAoAIAAgAH0ALAAKACAAIAB1AG4AZABlAGYAaQBuAGUAZAAKACkAOwA%3D&ruleDomains.project=all

### Expected result

For the following code:

```ts
const dates: Date[] = [];

export const latestDate = dates.reduce<Date | undefined>(
  (latest, current) => {
    if (!latest) {
      return current;
    }
    if (!current) {
      return latest;
    }
    return latest > current ? latest : current;
  },
  undefined
);
```

Expected behavior:
- There should be no linter diagnostics for `useMathMinMax`.

Actual behavior:
- The linter suggests to change `latest > current ? latest : current` to `Math.min(latest, current)`, which would be a breaking change. The original code produced a `latestDate` value with type `Date`, but the Biome-autofixed code produces a `number`.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by dyc3** [comment]
The fix is marked as unsafe, which means its up to the user to determine if the fix is OK, precisely for this reason.

We don't want this rule to require type information, and we don't currently have a mechanism to optionally use type information if its available.

In this case, you can simply suppress the rule
